composables/use-issues-by-column.ts

```diff
@@ -0,0 +1,40 @@
+import type { ColumnTitle } from '~/constants/columns'
+import type { IssueStatus } from '~/constants/issues'
+import type { Issue } from '~/types/issues'
+
+function filterIssuesWithoutStatus(issues: Issue[]): Issue[] {
+  const issuesWithoutStatus = issues.filter(
+    (issue) =>
+      !issue.labels.some((label) => label.startsWith(createPartialDataLabel('status'))),
+  )
+
+  return issuesWithoutStatus
+}
+
+function filterIssuesByStatus(issues: Issue[], status: IssueStatus): Issue[] {
+  const issuesWithStatus = issues.filter((issue) =>
+    issue.labels.includes(createDataLabel('status', status)),
+  )
+
+  return issuesWithStatus
+}
+
+function useIssuesByColumn(): Record<ColumnTitle, ComputedRef<Issue[]>> {
+  const route = useRoute('node-rid')
+  const response = useHttpdFetch('/projects/{rid}/issues', {
+    path: {
+      rid: route.params.rid,
+    },
+  })
+
+  const issues = response.data as Ref<Issue[] | undefined>
+
+  return {
+    'not-planned': computed(() => filterIssuesWithoutStatus(issues.value ?? [])),
+    'todo': computed(() => filterIssuesByStatus(issues.value ?? [], 'todo')),
+    'doing': computed(() => filterIssuesByStatus(issues.value ?? [], 'doing')),
+    'done': computed(() => filterIssuesByStatus(issues.value ?? [], 'done')),
```

Comment:
I think we should have a fundamentally different approach to resolving those. So as to

1. not have hard-coded columns we're looking for
2. have better algorithmic performance because we have nested loops across the whole collection in multiples here. This blows-up very quickly (roughly looks like `O(4 * CountOfIssues * CountOfLabels * cost of createLabel() * LengthOfLabel )`)
3. decrease complexity by not having this (admittedly messy) special "partial" label typings and functions

I was thinking of searching once to resolve unique column names and groups in one pass
    - reduce issues into a dictionary with column-data-labels as keys which have as value an array with the issues with that label e.g.
    ```js
    const issuesByColumn = {
      non-planned: [...],
      todo: [...],
			// ...
    }
    ```

Reply:
I'd like to resolve the first point before proceeding with the rest here

Reply:
hard-coded columns please. 

performance and algorithmic complexity are not concerns for this prototype.